onward/app/weather/WeatherApi.scala

```diff
@@ -36,7 +38,20 @@ object WeatherApi extends ExecutionContexts with ResourcesHelper {
     if (Play.isTest) {
       Future(Json.parse(slurpOrDie(new URI(url).getPath.stripPrefix("/"))))
     } else {
-      WS.url(url).get().filter(_.status == 200).map(_.json)
+      if(conf.switches.Switches.RetryFailedAccuWeatherApiRequests.isSwitchedOn) {
+        val fetchRequest = () => WS.url(url).withRequestTimeout(300).get().filter(_.status == 200).map { response =>
+          Right(response.json)
+        }
+        .recover {
+          case t : TimeoutException => Left(t)
+        }
+        retry.Backoff(max = 3, delay = Duration(100, TimeUnit.MILLISECONDS), base = 2)(fetchRequest).map {
+          case Left(error) => throw error
```

Comment:
Is it possible to `flatMap` and transform to  `case Left(error) => Future.failed(error)` instead of throwing the error?